File: src/main/java/de/byjoker/myjfql/database/DatabaseService.java
package de.byjoker.myjfql.database;

import de.byjoker.myjfql.util.StorageService;

import java.io.File;
import java.util.Collection;

public interface DatabaseService extends StorageService {

    void createDatabase(Database database);

    void saveDatabase(Database database);

    boolean existsDatabaseByIdentifier(String identifier);

    boolean existsDatabaseByName(String name);

    boolean existsDatabase(String id);

    void deleteDatabase(String id);

    Database loadDatabase(DatabaseType type, File file);

    Database getDatabaseByIdentifier(String identifier);

    Database getDatabaseByName(String name);

    Database getDatabase(String id);

    Collection<Database> getDatabases();

}

File: src/main/java/de/byjoker/myjfql/database/DatabaseService.kt
package de.byjoker.myjfql.database

import de.byjoker.myjfql.util.StorageService

interface DatabaseService : StorageService<Database> {

    fun createDatabase(database: Database)
    fun saveDatabase(database: Database)
    fun deleteDatabase(id: String)
    fun getDatabase(id: String): Database?
    fun getDatabaseByName(name: String): Database?
    fun getDatabaseByIdentifier(identifier: String): Database?
    fun getDatabases(): List<Database>

}
